examples/trivial_discord_bot.py

```diff
@@ -0,0 +1,148 @@
+# Remember when wiring random services together to see what you could make happen?
+# I miss those days
+
+# trivial discord bot implemented in mewbot - will respond to the message "!hello" with "World"
+
+from __future__ import annotations
+
+import json
+import logging
+import os
+
+from typing import Any, Dict, Set, Type
+
+from mewbot.bot import Bot
+
+from mewbot.behaviour import Behaviour
+from mewbot.api.v1 import Trigger, Condition, Action
+from mewbot.core import InputEvent, OutputEvent, OutputQueue
+
+from mewbot.http_post import PostIOConfig
+from mewbot.discord import DiscordInput, DiscordInputEvent, DiscordOutputEvent, DiscordOutput
+
+__location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+
+
+class DiscordTrivialTrigger(Trigger):
+    """
+    Nothing fancy - just fires whenever there is a DiscordInputEvent
+    """
+
+    @staticmethod
+    def consumes_inputs() -> Set[Type[InputEvent]]:
+        return {
+            DiscordInputEvent,
+        }
+
+    def matches(self, event: InputEvent) -> bool:
+        if not isinstance(event, DiscordInputEvent):
+            return False
+
+        return event.text == "!hello"
+
+
+class DiscordTrivialCondition(Condition):
+    """
+    Allows every input.
+    """
+
+    @staticmethod
+    def consumes_inputs() -> Set[Type[InputEvent]]:
+        return {
+            DiscordInputEvent,
+        }
+
+    def allows(self, event: InputEvent) -> bool:
+        return True
+
+
+class DiscordPrintAndReplyAction(Action):
+    """
+    Print every InputEvent.
+    """
+
+    _queue: OutputQueue
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.logger = logging.getLogger(__name__ + type(self).__name__)
+
+    @staticmethod
+    def consumes_inputs() -> Set[Type[InputEvent]]:
+        return {
+            DiscordInputEvent,
+        }
+
+    @staticmethod
+    def produces_outputs() -> Set[Type[OutputEvent]]:
+        return {
+            DiscordOutputEvent,
+        }
+
+    async def act(self, event: InputEvent, state: Dict[str, Any]) -> None:
+        """
+        Construct a DiscordOutputEvent with the result of performing the calculation.
+        """
+        if not isinstance(event, DiscordInputEvent):
+            self.logger.warning("Received wrong event type %s", type(event))
+            return
+
+        self.logger.info("event.txt = %s", event.text)
+        test_event = DiscordOutputEvent(
+            text="world", message=event.message, use_message_channel=True
+        )
+
+        self.logger.info(test_event)
+        await self.send(test_event)
+
+    async def send(self, event: OutputEvent) -> None:
+        await self._queue.put(event)
+
+
+def load_secrets() -> Any:
```

Comment:
Future Thoughts: Should this end up being a default/inbuilt thing to mewbot?

Reply:
Secret storage in the configuration language (and the YAML backing store) is something I've not really got a solution for yet. Thoughts on that welcome in the design docs.

Reply:
Will also consider it! That's definitely a problem that has to be solved.